server/core/src/irods_signal.cpp

```diff
@@ -1,38 +1,87 @@
 #include "irods_signal.hpp"
-#include "irods_stacktrace.hpp"
-#include "irods_logger.hpp"
 
-#include <string.h>
-#include <signal.h>
+#include "irods_default_paths.hpp"
 
+#include <boost/stacktrace/safe_dump_to.hpp>
+
+#include <csignal>
+#include <cstdio>
 #include <cstdlib>
-#include <sstream>
+#include <cstring>
+#include <ctime>
+#include <system_error>
+#include <charconv>
 
-// Define signal handlers for irods
+#include <unistd.h>
 
 extern "C"
+void stacktrace_signal_handler(int _signal)
 {
-    /// @brief Signal handler for seg faults
-    static void segv_handler(int _signal)
-    {
-        using log = irods::experimental::log;
+    timespec ts;
+
+    if (clock_gettime(CLOCK_REALTIME, &ts) != 0) {
+        std::exit(_signal);
+    }
+
+    const auto max_path_length = 512;
+
+    char file_path_hidden[max_path_length] = "../../";
+    std::strcat(file_path_hidden, IRODS_DEFAULT_PATH_HOMEDIR);
```

Comment:
why hidden?

Reply:
The hidden path keeps the stacktrace processor thread from touching it until the file is ready.

Reply:
oh, it's hidden from the processor... hmmm...   perhaps... `file_path_unprocessed`?

Reply:
that works.